### ISSUE TITLE
make at least one of WebSocket and RawSocket required in BP

### ISSUE THREAD

**Issue by goeddea** [wamp-proto/wamp-proto#226]
In principle, we do not want to require any specific transport for a WAMP implementation, but we do want to assure interoperability between WAMP client libs and routers. 

Stating that client libs are required to implement at a minimum one of WebSocket and RawSocket, and that routers are required to implement both achieves this.

see also  - https://groups.google.com/forum/#!msg/wampws/fBipuARHx6g/oFS2BQ-wBAAJ

**Comment by oberstet** [comment]
I don't agree: there might be implementations that only have function-based transport (when the library is actually piggy-backing on a real transport of an enclosing library), or where the transport is a message based IPC mechanism (such as, Netmap).

It is obvious that a WAMP implementation without _any_ transport is void of use.

_Requiring_ any specific transport is limiting ..

**Comment by gammazero** [comment]
I agree with @oberstet here.  There may be routers and clients that use WAMP over some transport other than WS or RS, and are perfectly suited for such an environment.  That should not mean that they are not WAMP-compliant.  There may even be a proxy that bridges such clients or servers to a socket transport, even though the WAMP server or clients do not operate in an environment with sockets.

OK, maybe rather contrived, but I feel that choice of transports should not be specified in WAMP spec.  If could be suggested that those are the most common transports and will offer the most interoperability with other implementations.
